pwndbg/dbg/lldb/repl/fuzzy.py

```diff
@@ -0,0 +1,307 @@
+            f.close()
+        self.join()
+
+
+class LLDBHistory(FileHistory):
+    """
+    Manage your LLDB History
+    """
+
+    def __init__(self, filename: str, ignore_duplicates: bool = False) -> None:
+        self.ignore_duplicates = ignore_duplicates
+        super().__init__(filename=filename)
+
+    def load_history_strings(self) -> list[str]:
+        strings = []
+        if os.path.exists(self.filename):
+            with open(self.filename) as f:
+                for string in reversed(f.read().splitlines()):
+                    if self.ignore_duplicates and string in strings:
+                        continue
+                    if string:
+                        strings.append(string)
+        return strings
+
+    def store_string(self, string: str) -> None:
+        with open(self.filename, "a") as f:
+            f.write(string.strip() + "\n")
+
+
+class LLDBCompleter(Completer):
+    """
+    Completer of LLDB
+    """
+
+    def __init__(self, dbg):
+        super().__init__()
+        self.dbg = dbg
+
+    def get_completions(
+        self, document: Document, complete_event: CompleteEvent
+    ) -> Iterator[Completion]:
+        target_text = document.text_before_cursor.lstrip()  # Ignore leading whitespaces
+
+        cursor_idx_in_completion = len(target_text)
+        all_completions, should_get_all_help_docs = get_lldb_completion_and_status(target_text, self.dbg)
+        if not all_completions:
+            return
+
+        for completion in all_completions:
+            if not completion.startswith(target_text):
+                continue
+            display_meta = (
+                None if not should_get_all_help_docs else safe_get_help_docs(completion) or None
+            )
+            # remove some prefix of raw completion
+            completion = completion[cursor_idx_in_completion:]
+            # display readable completion based on the text before cursor
+            display = re.split(r"\W+", target_text)[-1] + completion
+            yield Completion(completion, display=display, display_meta=display_meta)
+
+
+def wrap_with_history(function: Callable) -> Callable:
+    @functools.wraps(function)
+    def _wrapped(*a, **kw):
+        return function(*a, **kw)
+    return _wrapped
+
+
+def get_prompt_session(dbg):
+    bindings = KeyBindings()
+
+    if HAS_FZF:
+        # key binding for fzf history search
+        bindings.add("c-r")(fzf_reverse_search)
+        # key binding for fzf tab completion
+        fifo_in, fifo_out = create_preview_fifos()
+        preview = "echo {n} > %s\ncat %s" % (fifo_in, fifo_out)
+        bindings.add("c-i")(
+            functools.partial(
+                fzf_tab_autocomplete,
+                dbg=dbg,
+                preview=preview,
+                fifo_in=fifo_in,
+                fifo_out=fifo_out,
+            )
+        )
+    else:
+        print("Install fzf for better experience with pwndbg")
```

Comment:
I don't get what the need for this message is. Plus, it seems this function is never called unless `HAZ_FZF` is true, anyway. Maybe take this out?
```suggestion
    # key binding for fzf history search
    bindings.add("c-r")(fzf_reverse_search)
    # key binding for fzf tab completion
    fifo_in, fifo_out = create_preview_fifos()
    preview = "echo {n} > %s\ncat %s" % (fifo_in, fifo_out)
    bindings.add("c-i")(
        functools.partial(
            fzf_tab_autocomplete,
            dbg=dbg,
            preview=preview,
            fifo_in=fifo_in,
            fifo_out=fifo_out,
        )
    )
```

Reply:
Accepted.